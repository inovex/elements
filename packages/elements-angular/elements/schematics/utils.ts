import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
