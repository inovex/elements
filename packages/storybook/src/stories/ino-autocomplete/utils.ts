import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
