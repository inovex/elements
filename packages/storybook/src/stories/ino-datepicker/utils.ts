import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
