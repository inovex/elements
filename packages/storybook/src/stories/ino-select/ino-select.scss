.story-select {

  // Ensure that the select can collapse within the story borders.
  min-height: 200px;
  
  ino-select {
    margin-bottom: 12px;
  }
}
