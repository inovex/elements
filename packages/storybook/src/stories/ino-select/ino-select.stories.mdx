import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-select/readme.md";

<Meta title="Input/ino-select/Documentation" />

<Description markdown={readme} />
