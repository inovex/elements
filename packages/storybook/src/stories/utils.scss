/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
