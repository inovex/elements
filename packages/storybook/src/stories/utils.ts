import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
