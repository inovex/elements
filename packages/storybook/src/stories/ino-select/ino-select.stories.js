import { boolean, text } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-select/readme.md';
import optionComponentReadme from '_local-elements/src/components/ino-option/readme.md';
import optionGroupComponentReadme from '_local-elements/src/components/ino-option-group/readme.md';

import './ino-select.scss';

// https://github.com/storybooks/storybook/issues/4337#issuecomment-428495664
function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function(e) {
    const el = e.target;
    if (el.tagName.toLowerCase() !== 'ino-select') {
      return;
    }

    el.setAttribute('value', e.detail);
  };

  document.addEventListener('valueChange', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('valueChange', eventHandler);
  };
}

export default {
  title: 'Input/<ino-select>',
  parameters: {
    actions: {
      handles: ['valueChange .customizable-select', 'submit .form']
    }
  },
  decorators: [
    story => {
      subscribeToComponentEvents();
      return story();
    }
  ]
};

export const DefaultUsage = () => {
  const optionsTemplate = /*html*/ `
    <ino-option value="Option 1">Option 1</ino-option>
    <ino-option value="Option 2">Option 2</ino-option>
    <ino-option value="Option 3">Option 3</ino-option>
    <ino-option value="Option 4">Option 4</ino-option>
    `;

  return /*html*/ `
      <div class="story-select">
        <ino-select class="customizable-select"
          name="${text('name', 'select-1')}"
          ino-outline="${boolean('ino-outline', false)}"
          ino-label="${text('ino-label', 'Customizable select')}"
          required="${boolean('required', false)}"
          ino-show-label-hint="${boolean('ino-show-label-hint', false)}"
          value="${text('value', 'Option 1')}"
        >
          ${optionsTemplate}
        </ino-select>

        <ino-select ino-label="Select with Groups">
            <ino-option-group ino-label="My First Group">
                ${optionsTemplate}
            </ino-option-group>
            <ino-option-group ino-label="My Second Group">
                <ino-option value="Option 5">Option 5</ino-option>
                <ino-option value="Option 6">Option 6</ino-option>
                <ino-option value="Option 7">Option 7</ino-option>
                <ino-option value="Option 8">Option 8</ino-option>
            </ino-option-group>
          </ino-select>

        <ino-select style="--select-height: 80px" ino-label="Select with custom height">
          ${optionsTemplate}
        </ino-select>

        <h4 class="outline">Outlined</h4>
        <ino-select ino-outline>
          ${optionsTemplate}
          <ino-option value="Option 5" selected>Option 5</ino-option>
        </ino-select>

        <h4>Labels</h4>
        <ino-select>
          <ino-option value="No Label" selected>No label</ino-option>
          ${optionsTemplate}
        </ino-select>
        <ino-select ino-label="Floating label">
         <ino-option value="Selected Option" selected>Selected Option</ino-option>
          ${optionsTemplate}
        </ino-select>

        <ino-select ino-label="Floating label outlined" ino-outline>
          <ino-option value="Selected Option" selected>Selected Option</ino-option>
          ${optionsTemplate}
        </ino-select>

        <h4>States</h4>
        <ino-select disabled ino-label="Disabled select">
          <ino-option value="Selected Option" selected>Selected Option</ino-option>
          ${optionsTemplate}
        </ino-select>
        <ino-select ino-label="Select with disabled option">
          ${optionsTemplate}
          <ino-option value="Disabled Option" disabled>Disabled Option</ino-option>
        </ino-select>

        <ino-select required ino-label="required select" ino-show-label-hint>
          ${optionsTemplate}
        </ino-select>

        <ino-select ino-label="optional select" ino-show-label-hint>
          ${optionsTemplate}
        </ino-select>
      </div>
    `;
};

DefaultUsage.decorators = [withStencilReadme(componentReadme)];


export const Forms = () => /*html*/ `
<div class="story-select">
  <h4>Required</h4>
  <p>The form should not submit since no option is selected and the select is required.</p>
  <form class="form" onSubmit="alert('Form submitted'); return false;">
      <ino-select ino-label="Form select" required>
        <ino-option value="Test">Test</ino-option>
      </ino-select>
      <ino-button type="submit">Submit</ino-button>
  </form>
</div>
`;

export const SelectOptions = () => /*html*/ `
    <style>
        ino-option.customizable-option {
          --ino-option-selected-background-color: ${text('--ino-option-selected-background-color', 'rgba(61, 64, 245, 0.05)', 'Custom Properties')};
          --ino-option-selected-background-color-hover: ${text('--ino-option-selected-background-color-hover', 'rgba(61, 64, 245, 0.1)', 'Custom Properties')};
          --ino-option-selected-background-color-focus: ${text('--ino-option-selected-background-color-focus', 'rgba(61, 64, 245, 0.15)', 'Custom Properties')};
          --ino-option-selected-background-color-active: ${text('--ino-option-selected-background-color-active', 'rgba(61, 64, 245, 0.3)', 'Custom Properties')};
          --ino-option-deselected-background-color: ${text('--ino-option-deselected-background-color', '', 'Custom Properties')};
          --ino-option-deselected-background-color-hover: ${text('--ino-option-deselected-background-color-hover', '', 'Custom Properties')};
          --ino-option-deselected-background-color-focus: ${text('--ino-option-deselected-background-color-focus', '', 'Custom Properties')};
          --ino-option-deselected-background-color-active: ${text('--ino-option-deselected-background-color-active', '', 'Custom Properties')};
        }
    </style>
    <h4>Customizable option</h4>
    <ino-select ino-label="Customizable option">
        <ino-option
            class="customizable-option"
            disabled="${boolean('disabled', false)}"
            value="${text('value', 'Content')}"
        >
            Content
        </ino-option>
        <ino-option
            class="customizable-option"
            disabled="${boolean('disabled', false)}"
            value="${text('value', 'Content')}"
        >
            Content
        </ino-option>
    </ino-select>

    <h4>Disabled option</h4>
    <ino-select ino-label="Disabled option">
        <ino-option value="Content" disabled>Content</ino-option>
    </ino-select>

    <h4>Selected option</h4>
        <ino-select ino-label="Selected option">
        <ino-option value="Content" selected>Content</ino-option>
    </ino-select>
  `;

SelectOptions.decorators = [withStencilReadme(optionComponentReadme)];

export const OptionGroup = () => /*html*/ `
    <h4>Option Group Example</h4>
    <ino-select ino-label="Select with Groups">
      <ino-option-group ino-label="My First Group">
          <ino-option value="Option 1">Option 1</ino-option>
          <ino-option value="Option 2">Option 2</ino-option>
          <ino-option value="Option 3">Option 3</ino-option>
          <ino-option value="Option 4">Option 4</ino-option>
      </ino-option-group>
      <ino-option-group ino-label="My Second Group">
          <ino-option value="Option 5">Option 5</ino-option>
          <ino-option value="Option 6">Option 6</ino-option>
          <ino-option value="Option 7">Option 7</ino-option>
          <ino-option value="Option 8">Option 8</ino-option>
      </ino-option-group>
    </ino-select>
  `;

OptionGroup.decorators = [withStencilReadme(optionGroupComponentReadme)];
