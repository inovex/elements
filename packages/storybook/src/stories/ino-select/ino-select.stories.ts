import { Components } from '@inovex.de/elements';
import { html } from 'lit-html';
import { useEffect } from '@storybook/client-api';
import { defaultDecorator } from '../utils';
import './ino-select.scss';
import { Story } from '@storybook/web-components';

export default {
  title: 'Input/ino-select',
  component: 'ino-select',
  parameters: {
    actions: {
      handles: ['valueChange .customizable-select', 'submit .form'],
    },
  },
  decorators: [
    story => defaultDecorator(story, 'story-select'), 
    story => {
      useEffect(() => {
        const eventHandler = e => e.target.setAttribute('value', e.detail);
        const selects = document.querySelectorAll('ino-select');
        selects.forEach(s => s.addEventListener('valueChange', eventHandler));
        return () => selects.forEach(s => s.removeEventListener('valueChange', eventHandler));
      });
      return story();
    }
  ],
};

const optionsTemplate = html`
  <ino-option value="Option 1">Option 1</ino-option>
  <ino-option value="Option 2">Option 2</ino-option>
  <ino-option value="Option 3">Option 3</ino-option>
`;

export const Playground: Story<Components.InoSelect> = args => html`
  <ino-select
    disabled="${args.disabled}"
    name="${args.name}"
    outline="${args.outline}"
    label="${args.label}"
    required="${args.required}"
    show-label-hint="${args.showLabelHint}"
    value="${args.value}"
  >
    ${optionsTemplate}
  </ino-select>
`;
Playground.args = {
  disabled: false,
  label: 'Select label',
  name: 'select-1',
  outline: false,
  required: false,
  showLabelHint: false,
  value: 'Option 1',
}

export const NoLabel = () => html`
  <ino-select>
    <ino-option value="No Label" selected>Selected option</ino-option>
    ${optionsTemplate}
  </ino-select>
`;

export const FloatingLabel = () => html`
  <ino-select label="Floating label">
    <ino-option value="Selected Option" selected>Selected Option</ino-option>
    ${optionsTemplate}
  </ino-select>
`;

export const OutlineFloatingLabel = () => html`
  <ino-select label="Floating label outlined" outline>
    <ino-option value="Selected Option" selected>Selected Option</ino-option>
    ${optionsTemplate}
  </ino-select>
`;

export const Disabled = () => html`
  <ino-select disabled label="Disabled select">
    <ino-option value="Selected Option" selected>Selected Option</ino-option>
    ${optionsTemplate}
  </ino-select>
`; 

export const DisabledOption = () => html`
   <ino-select label="Select with disabled option">
    ${optionsTemplate}
    <ino-option value="Disabled Option" disabled>Disabled Option</ino-option>
  </ino-select>
`;

export const Required = () => html`
  <ino-select required label="Required select" show-label-hint>
    ${optionsTemplate}
  </ino-select>
`;

export const Option = () => html`
  <ino-select label="Optional select" show-label-hint>
    ${optionsTemplate}
  </ino-select>
`;

// TODO
/*<style>
        ino-option.customizable-option {
          --ino-option-selected-background-color: ${text(
            '--ino-option-selected-background-color',
            'rgba(61, 64, 245, 0.05)',
            'Custom Properties'
          )};
          --ino-option-selected-background-color-hover: ${text(
            '--ino-option-selected-background-color-hover',
            'rgba(61, 64, 245, 0.1)',
            'Custom Properties'
          )};
          --ino-option-selected-background-color-focus: ${text(
            '--ino-option-selected-background-color-focus',
            'rgba(61, 64, 245, 0.15)',
            'Custom Properties'
          )};
          --ino-option-selected-background-color-active: ${text(
            '--ino-option-selected-background-color-active',
            'rgba(61, 64, 245, 0.3)',
            'Custom Properties'
          )};
          --ino-option-deselected-background-color: ${text(
            '--ino-option-deselected-background-color',
            '',
            'Custom Properties'
          )};
          --ino-option-deselected-background-color-hover: ${text(
            '--ino-option-deselected-background-color-hover',
            '',
            'Custom Properties'
          )};
          --ino-option-deselected-background-color-focus: ${text(
            '--ino-option-deselected-background-color-focus',
            '',
            'Custom Properties'
          )};
          --ino-option-deselected-background-color-active: ${text(
            '--ino-option-deselected-background-color-active',
            '',
            'Custom Properties'
          )};
        }
    </style>
*/
// Does not work as described
/* export const Forms = () => html`
  <div class="story-select">
    <p>The form should not submit since no option is selected and the select is required.</p>
    <form class="form" onSubmit="alert('Form submitted'); return false;">
        <ino-select label="Form select" required>
          <ino-option value="Test">Test</ino-option>
        </ino-select>
        <button type="submit">Submit</button>
    </form>
  </div>
`; */
