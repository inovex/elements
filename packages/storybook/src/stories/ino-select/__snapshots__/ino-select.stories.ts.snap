// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-select Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-select">
    <ino-select disabled="true"
                name="select-1"
                outline="false"
                label="Select label"
                required="false"
                show-label-hint="false"
                value="Option 1"
                error="false"
                helper
                helper-persistent="false"
                helper-validation="false"
    >
      <ino-option value="Option 1">
        Option 1
      </ino-option>
      <ino-option value="Option 2">
        Option 2
      </ino-option>
      <ino-option value="Option 3">
        Option 3
      </ino-option>
    </ino-select>
  </div>
</div>
`;

exports[`Input/ino-select Error smoke-test 1`] = `
<div id="root-inner">
  <div class="story-select">
    <ino-select disabled="false"
                name="select-1"
                outline="false"
                label="Select label"
                required="false"
                show-label-hint="false"
                value="Option 1"
                error="true"
                helper
                helper-persistent="false"
                helper-validation="false"
    >
      <ino-option value="Option 1">
        Option 1
      </ino-option>
      <ino-option value="Option 2">
        Option 2
      </ino-option>
      <ino-option value="Option 3">
        Option 3
      </ino-option>
    </ino-select>
  </div>
</div>
`;

exports[`Input/ino-select Form smoke-test 1`] = `
<div id="root-inner">
  <div class="story-select">
    <form>
      <p>
        Form should not submit if no value is selected
      </p>
      <ino-select required>
        <ino-option value="Option 1">
          Option 1
        </ino-option>
        <ino-option value="Option 2">
          Option 2
        </ino-option>
        <ino-option value="Option 3">
          Option 3
        </ino-option>
      </ino-select>
      <ino-button type="submit">
        Submit
      </ino-button>
    </form>
  </div>
</div>
`;

exports[`Input/ino-select HelperMessage smoke-test 1`] = `
<div id="root-inner">
  <div class="story-select">
    <ino-select disabled="false"
                name="select-1"
                outline="false"
                label="Select label"
                required="false"
                show-label-hint="false"
                value="Option 1"
                error="false"
                helper="My Helper Message"
                helper-persistent="true"
                helper-validation="false"
    >
      <ino-option value="Option 1">
        Option 1
      </ino-option>
      <ino-option value="Option 2">
        Option 2
      </ino-option>
      <ino-option value="Option 3">
        Option 3
      </ino-option>
    </ino-select>
  </div>
</div>
`;

exports[`Input/ino-select HelperMessageValidation smoke-test 1`] = `
<div id="root-inner">
  <div class="story-select">
    <ino-select disabled="false"
                name="select-1"
                outline="false"
                label="Select label"
                required="true"
                show-label-hint="false"
                value="null"
                error="false"
                helper="This message will be highlighted when no option has been selected"
                helper-persistent="true"
                helper-validation="true"
    >
      <ino-option value="Option 1">
        Option 1
      </ino-option>
      <ino-option value="Option 2">
        Option 2
      </ino-option>
      <ino-option value="Option 3">
        Option 3
      </ino-option>
    </ino-select>
  </div>
</div>
`;

exports[`Input/ino-select Outlined smoke-test 1`] = `
<div id="root-inner">
  <div class="story-select">
    <ino-select disabled="false"
                name="select-1"
                outline="true"
                label="Select label"
                required="false"
                show-label-hint="false"
                value="Option 1"
                error="false"
                helper
                helper-persistent="false"
                helper-validation="false"
    >
      <ino-option value="Option 1">
        Option 1
      </ino-option>
      <ino-option value="Option 2">
        Option 2
      </ino-option>
      <ino-option value="Option 3">
        Option 3
      </ino-option>
    </ino-select>
  </div>
</div>
`;

exports[`Input/ino-select Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-select">
    <ino-select disabled="false"
                name="select-1"
                outline="false"
                label="Select label"
                required="false"
                show-label-hint="false"
                value="Option 1"
                error="false"
                helper
                helper-persistent="false"
                helper-validation="false"
    >
      <ino-option value="Option 1">
        Option 1
      </ino-option>
      <ino-option value="Option 2">
        Option 2
      </ino-option>
      <ino-option value="Option 3">
        Option 3
      </ino-option>
    </ino-select>
  </div>
</div>
`;

exports[`Input/ino-select ShowLabelHint smoke-test 1`] = `
<div id="root-inner">
  <div class="story-select">
    <ino-select disabled="false"
                name="select-1"
                outline="false"
                label="Select label"
                required="false"
                show-label-hint="true"
                value="Option 1"
                error="false"
                helper
                helper-persistent="false"
                helper-validation="false"
    >
      <ino-option value="Option 1">
        Option 1
      </ino-option>
      <ino-option value="Option 2">
        Option 2
      </ino-option>
      <ino-option value="Option 3">
        Option 3
      </ino-option>
    </ino-select>
  </div>
</div>
`;

exports[`Input/ino-select WithIcon smoke-test 1`] = `
<div id="root-inner">
  <div class="story-select">
    <div style="height: 400px;">
      <ino-select label="Select with leading icon">
        <ino-icon slot="icon-leading"
                  icon="user"
        >
        </ino-icon>
        <ino-option value="Selected Option"
                    selected
        >
          Selected Option
        </ino-option>
        <ino-option value="Option 1">
          Option 1
        </ino-option>
        <ino-option value="Option 2">
          Option 2
        </ino-option>
        <ino-option value="Option 3">
          Option 3
        </ino-option>
      </ino-select>
      <ino-select style="margin-top: 150px;"
                  outline
                  label="Select with leading icon"
      >
        <ino-icon slot="icon-leading"
                  icon="user"
        >
        </ino-icon>
        <ino-option value="Selected Option"
                    selected
        >
          Selected Option
        </ino-option>
        <ino-option value="Option 1">
          Option 1
        </ino-option>
        <ino-option value="Option 2">
          Option 2
        </ino-option>
        <ino-option value="Option 3">
          Option 3
        </ino-option>
      </ino-select>
    </div>
  </div>
</div>
`;
